src/ensureChildrenIsValid.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+export const errorMessage =
+  '<Waypoint> expected to receive a function or a single React element child.\n\n' +
+  'See /* TODO: shortlink to Children section */ for more info.';
+
+/**
+ * Raise an error if "children" is not a function or if more that one child was provided
+ *
+ * @param {?(React.element|Function)} children
+ * @return {undefined}
+ */
+export default function ensureChildrenIsValid(children) {
```

Comment:
With the added `innerRef` support, I don't think we should support the function as a child as well (since they solve basically the same usecase).

Reply:
Fair point, I'll remove it